Ecom/GoCart/CartFunctionalityViewController.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  functionalityView.swift
+//  GoCart
+//
+//  Created by Devasurya on 27/12/23.
+//
+
+import UIKit
+
+/// Protocoal- for transferring data to tableview.
+protocol AddDataToMainVC{
+    func data(result:Product)
+}
+
+/// FunctionalityView is the view responsilbe for the user action to select an image using a pickerview.
+class CartFunctionalityViewController: UIViewController ,UIImagePickerControllerDelegate,UINavigationControllerDelegate {
+    
+    /// Referencing Outlet.
+    @IBOutlet weak var iconText: UITextField!
+    @IBOutlet weak var cartImage: UIImageView!
+  
+    /// Variable declaration.
+    var delegate: AddDataToMainVC?
+    
+    override func viewDidLoad() {
+        
+    }
+
+    /// Button Action to  Upload image.
+    /// - Parameter sender: UIbutton
+    @IBAction func uploadProductImage(_ sender: UIButton) {
+        uploadProductImage()
+    }
+    
+    /// didFinishPickingMediaWithInfo is  a built in method in UIImagePickerController to set the result into a constant and asigning that to the UIelement.
+    /// - Parameters:
```

Comment:
move it to extension Imagepickercontroller . also the documentation parameters are missing